client/chain-spec/src/chain_spec.rs

```diff
@@ -76,13 +76,13 @@ impl<G: RuntimeGenesis, E> BuildStorage for ChainSpec<G, E> {
 			Genesis::Runtime(gc) => gc.build_storage(),
 			Genesis::Raw(RawGenesis { top: map, children: children_map }) => Ok(Storage {
 				top: map.into_iter().map(|(k, v)| (k.0, v.0)).collect(),
-				children: children_map.into_iter().map(|(sk, child_content)| {
+				children: children_map.into_iter().map(|(storage_key, child_content)| {
```

Comment:
I agree on all your point, my first idea was simply new type of child trie then new field in genesis raw for the new child trie.
So I guess renaming 'children' to 'children_default' would be what makes more sense (it does not break kusama genesis because the json do not name the field (it is using an old format that is accepted (array as a map that was something I noticed last time I change the json)).
But adding new field is certainly breaking as the command for generating genesis does explicitely name the field in the json (I will need to see if there is a way to force the array syntax).